src/components/UserProfile/UserSettings/UserGeneralSettings/index.tsx

```diff
@@ -206,6 +217,26 @@ const UserGeneralSettings: React.FC = () => {
                   )}
                 </div>
               </div>
+              <div className="form-row">
+                <label htmlFor="minimumAvailability" className="text-label">
+                  {intl.formatMessage(messages.applanguage, {
+                    applicationTitle: currentSettings.applicationTitle,
+                  })}
+                </label>
+                <div className="form-input">
+                  <div className="form-input-field">
+                    <Field as="select" id="locale" name="locale">
+                      {(Object.keys(
+                        availableLanguages
+                      ) as (keyof typeof availableLanguages)[]).map((key) => (
+                        <option key={key} value={availableLanguages[key].code}>
+                          {availableLanguages[key].display}
+                        </option>
+                      ))}
+                    </Field>
+                  </div>
+                </div>
+              </div>
```

Comment:
We should only show this setting when modifying one's own user settings. Also, do we want to make a note that it is only for the current device? Alternatively, it wouldn't be a bad idea to transition this to a per-user global setting with this change, since I don't really see a use case where a user would want to use different languages on different devices.

Reply:
I want to transition this to a user setting but its actually a ton of work so I was trying to refrain from piling it into this PR.

Moving it into settings means the client no longer has to pass the locale to the API and we can have it on the server-side ready to go. It's a lot of refactoring. I'll consider including it :)